paasta_tools/cli/cmds/spark_run.py

```diff
@@ -1150,6 +1155,40 @@ def paasta_spark_run(args):
         needs_docker_cfg=needs_docker_cfg,
         auto_set_temporary_credentials_provider=auto_set_temporary_credentials_provider,
     )
+
+    # Mass enable DRA config
+    if (
+        args.cluster_manager == CLUSTER_MANAGER_K8S
+        and "spark.dynamicAllocation.enabled" not in spark_conf
+    ):
+        num_executors = int(spark_conf["spark.executor.instances"])
+        for exec_range in EXECUTOR_RANGES:
+            if exec_range[0] <= num_executors <= exec_range[1]:
+
+                log.warn(
+                    PaastaColors.yellow(
+                        "Spark Dynamic Resource Allocation (DRA) enabled for this batch. "
+                        "More info: y/spark-dra. If you wish to disable DRA, please provide "
+                        "spark.dynamicAllocation.enabled=false in --spark-args\n"
+                    )
+                )
+                user_spark_opts["spark.dynamicAllocation.enabled"] = "true"
+                spark_conf = get_spark_conf(
```

Comment:
Should this be out of the 'if' statement?

Reply:
No this method of service-configuration-lib is being called to configure other DRA configs

Reply:
Expose the update_dra_config method from service-configuration-lib and use that